gpio-cdev: add nu801 userspace driver

riptidewave93 opened the pull request:
This adds a userspace interpretation of the nu801 driver used by Meraki
hardware. Previously this was a driver that was added per target, but as
multiple targets now have this driver, we should move to something that
can be shared by all targets since no driver exists upstream.

Co-developed-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Chris Blake <chrisrblake93@gmail.com>
Signed-off-by: Christian Lamparter <chunkeey@gmail.com>

This is a followup to the previous RFC PR that was at https://github.com/openwrt/openwrt/pull/4284. Now that 5.10 is the 
default kernel for x86, I am working on pre-reqs for getting Meraki
MX100 support merged into OpenWRT.

CC @chunkeey

adschm commented on package/system/gpio-cdev/nu801/Makefile:
```diff
@@ -0,0 +1,41 @@
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=nu801
```
This nested folder structure seems uncommon. Can you make it a package at the same level as all the others?
Or is this the beginning of a greater plan?

riptidewave93 replied:
@adschm TL;DR this is the first of it's kind, a userspace LED driver using CONFIG_GPIO_CDEV. Due to this, it felt appropriate to break it out into a new folder of gpio-cdev since more drivers like this may be added in the future. However, if you have a better idea on placement, I am open to suggestions.

adschm commented on package/system/gpio-cdev/nu801/Makefile:
```diff
@@ -0,0 +1,41 @@
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Userspace GPIO Drivers
+  DEPENDS:=@LINUX_5_10 +kmod-leds-uleds
```
@!LINUX_5_4 instead ...

riptidewave93 replied:
@adschm since this relies on CONFIG_GPIO_CDEV, which isn't available in 5.4, it needs to have a dependency on 5.10 or newer.

adschm replied:
Indeed, since your version does not cover the "or newer".

riptidewave93 committed:
gpio-cdev: add nu801 userspace driver

This adds a userspace interpretation of the nu801 driver used by Meraki
hardware. Previously this was a driver that was added per target, but as
multiple targets now have this driver, we should move to something that
can be shared by all targets since no driver exists upstream.

Co-developed-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Chris Blake <chrisrblake93@gmail.com>
Signed-off-by: Christian Lamparter <chunkeey@gmail.com>

riptidewave93 commented:
@adschm Just curious what's required to get this merged in at this point, thanks!

chunkeey commented:
I would like to see it being used for by on of the aforementioned devices.
Can't you please just make a PR for the MX100 and add this commit to the series?

Thank You!

riptidewave93 commented:
> I would like to see it being used for by on of the aforementioned devices. Can't you please just make a PR for the MX100 and add this commit to the series?
> 
> Thank You!

Ill go ahead and do this instead then, thanks!
